lib/src/repositories/ton_wallet_repository/ton_wallet_repository_impl.dart

```diff
@@ -225,15 +211,21 @@ mixin TonWalletRepositoryImpl implements TonWalletRepository {
 
   @override
   Future<void> updateTransportSubscriptions() async {
-    closeAllSubscriptions();
+    final assets = wallets
```

Comment:
I think, this place is a bottom neck, because if you change transport until all subscriptions were completed, you will lose some part of wallets, so imho for this place you must use some cache list as it was `lastUpdatedAddress` and you won't lose necessary wallets. But `lastUpdatedAddress` must be used only here, so when you will revert this, leave a comment with description.

Reply:
Also waiting for `TokenWalletRepositoryImpl` with the same logic